plugins/module_utils/firewall_rules_utils.py

```diff
@@ -280,60 +282,122 @@ def __post_init__(self):
+
+        Example:
+        Given a FirewallRule object with attributes like {"source_any": "1", "source_port": 22},
+        the output will be an XML element structured as:
+        ```xml
+        <rule>
+            <source>
+                <any/>
+                <port>22</port>
+            </source>
+        </rule>
+        ```
+
+        Note: The method assumes the presence of a utility function `dict_to_etree` for
+        converting dictionaries to XML elements.
+        """
+
         rule_dict: dict = asdict(self)
         del rule_dict["uuid"]
 
-        # Here data from the dataclass in this form eg.:
-        # { "source_any": "1" }
-        # into the required xml data structure:
-        # <source><any/></source>
         for direction in ["source", "destination"]:
             for key in ["address", "network", "port", "any", "not"]:
                 current_val: Optional[Any] = rule_dict.get(
                     f"{direction}_{key}"
                 )  # source_not = None
-                logging.info(f"{direction}_{key} : {current_val}")
                 if current_val is not None:
-
                     if rule_dict.get(direction) is None:
                         rule_dict[direction] = {}
 
                     # s/d_network
                     if isinstance(current_val, bool):
                         if current_val:
-                            # if current boolean value is 'True' it suffices
-                            # to write an empty tag into the xml:
-                            # e.g. {"any": True } == <any/>
-                            # no need for <any>1</any>
-
                             rule_dict[direction][key] = None
                     else:
-                        # every other option should be written into the
-                        # corresponding structure e.g:
-                        # { "source_port": 22 } == <source><port>22</port></source>
                         rule_dict[direction][key] = current_val
                 del rule_dict[f"{direction}_{key}"]
-        for rule_key, rule_val in rule_dict.copy().items():  # rule_val = "{"not":None}"
-            if rule_val is None or rule_val is False:
-                # remove unnecessary fields
+        for rule_key, rule_val in rule_dict.copy().items():
+            if (rule_val is None or rule_val is False) and rule_key != "quick":
                 del rule_dict[rule_key]
                 continue
             if issubclass(type(rule_val), ListEnum):
                 rule_dict[rule_key] = rule_val.value
+
+            elif rule_key == "quick":
+                if rule_val is True:
+                    del rule_dict["quick"]
+                else:
+                    rule_dict[rule_key] = "0"
+
             elif isinstance(rule_val, bool):
-                # assume it rule_val is True because of first check in the for loop
+                if "quick" in rule_key:
+                    raise Exception(rule_val)
```

Comment:
I would handle the 'quick" case as first such that we do not need to handle it further on. Eg.:
```python3
        for rule_key, rule_val in rule_dict.copy().items():
            if rule_key == "quick":
                if rule_val:
                    del rule_dict[rule_key]
                    continue
                rule_dict[rule_key] = "0"
            elif rule_val in [None, False]:
                del rule_dict[rule_key]
            elif issubclass(type(rule_val), ListEnum):
                rule_dict[rule_key] = rule_val.value
            elif isinstance(rule_val, bool):
                rule_dict[rule_key] = "1"
```

(Sorry GH does not allow me to insert a suggestion when there are deleted lines in the diff 🤷🏼‍♂️ )

Reply:
thanks for the suggestion! Added in [6b30813](https://github.com/DonGiovanni83/puzzle.opnsense/pull/3/commits/6b30813801ff9977e4be018ac4bf8f8b90151ac7).